app/models/manageiq/providers/ovirt/infra_manager.rb

```diff
@@ -42,7 +44,7 @@ class ManageIQ::Providers::Ovirt::InfraManager < ManageIQ::Providers::InfraManag
   end
 
   supports :create_iso_datastore do
-    unsupported_reason_add(:create_iso_datastore, _("Already has an ISO datastore")) if iso_datastore
+    unsupported_reason_add(:create_iso_datastore, _("Already has an ISO datastore")) if iso_datastores
```

Comment:
I think you want `if iso_datastores.any?` or `unless iso_datastores.empty?` since `[]` is going to return truthy